### PR TITLE
Add a performance regression issue template

### LINKED ISSUE #7821: Performance regression issue template
## What problem does this solve or what need does it fill?
There's a few old issues that are focusing on poor performance, but do not provide sufficient information on the bottleneck, which may end up causing it to be difficult or impossible to reproduce without the exact machine configuration the author is using. Examples of this include #6363, #3687, and #3052.

## What solution would you like?
A new issue template that tries to ask the author to provide even deeper investigations into the poor performance (i.e. Tracy/chrome traces). We have plenty of documentation on performance profiling, so directly linking them to it should at least get most of the issue authors to dig a bit deeper.

## What alternative(s) have you considered?
Extend the bug report issue template to do the same thing.

### PR DESCRIPTION
**Pull request by james7132**
# Objective
Closes #7821. A good chunk of performance related issues filed are tough to take action on without much deeper investigation by the author, who may or may not be responsive after filing the issue.

## Solution
Add a performance regression issue template and use it to direct users to take a more proactive role in investigating the source of the regression.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+## Bevy version
+
+* Original: The release number or commit hash of the version you last tested your app against.
+* Current: The release number or commit hash of the version you're currently using.
+
+## \[Optional\] Relevant system information
+
+If you cannot get Bevy to build or run on your machine, please include:
```
Comment (alice-i-cecile) [inline .github/ISSUE_TEMPLATE/performance_regression.md:16]: I don't think this belongs here. However it *does* suggest a need for a different template just for this perhaps.

Comment (james7132) [reply .github/ISSUE_TEMPLATE/performance_regression.md]: Removed the part about getting it to build, but we should definitely have a section for system information, as the OS and hardware used will definitely have an impact.

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  - nightly should generally work, but there are sometimes regressions: please let us know!
+- the operating system or browser used, including its version
+  - e.g. Windows 10, Ubuntu 18.04, iOS 14
+
+## What's performing poorly?
+
+Describe how you arrived at the problem. If you can, consider providing a code snippet or link
+to help reproduce the regression. 
+
+If the exact scenario is not immediately reproducible, please include a set list of steps
```
Comment (alice-i-cecile) [inline .github/ISSUE_TEMPLATE/performance_regression.md:29]:
```suggestion
If the exact scenario is not immediately reproducible on `cargo run`, please include a set list of steps to produce the correct setup.
```

**Review by alice-i-cecile — APPROVED** [review]

**Review by cart — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- the Rust version you're using (you can get this by running `cargo --version`)
+  - Bevy relies on the "latest stable release" of Rust
+  - nightly should generally work, but there are sometimes regressions: please let us know!
+- the operating system or browser used, including its version
+  - e.g. Windows 10, Ubuntu 18.04, iOS 14
+
+## What's performing poorly?
+
+Describe how you arrived at the problem. If you can, consider providing a code snippet or link
+to help reproduce the regression.
```
Comment (mockersf) [inline .github/ISSUE_TEMPLATE/performance_regression.md:27]:
```suggestion
to help reproduce the regression.
```

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+## What's performing poorly?
+
+Describe how you arrived at the problem. If you can, consider providing a code snippet or link
+to help reproduce the regression. 
+
+If the exact scenario is not immediately reproducible on `cargo run`, please include a set list of steps to produce the correct setup.
+
+## Before and After Traces
+
+To best help us investigate the regression, it's best to provide as much detailed profiling
```
Comment (mockersf) [inline .github/ISSUE_TEMPLATE/performance_regression.md:33]:
```suggestion
To best help us investigate the regression, it's best to provide as much detailed profiling
```

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+to help reproduce the regression. 
+
+If the exact scenario is not immediately reproducible on `cargo run`, please include a set list of steps to produce the correct setup.
+
+## Before and After Traces
+
+To best help us investigate the regression, it's best to provide as much detailed profiling 
+data as possible.
+
+If your app is running slowly, please show profiler traces before and after the change.
```
Comment (mockersf) [inline .github/ISSUE_TEMPLATE/performance_regression.md:36]:
```suggestion
If your app is running slowly, please show profiler traces before and after the change.
```

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+## Additional information
+
+Other information that can be used to further reproduce or isolate the problem.
+This commonly includes:
+
+- screenshots
+- logs
+- theories about what might be going wrong
+- workarounds that you used
+- links to related bugs, PRs or discussions
```
Comment (mockersf) [inline .github/ISSUE_TEMPLATE/performance_regression.md:55]:
```suggestion
* screenshots
* logs
* theories about what might be going wrong
* workarounds that you used
* links to related bugs, PRs or discussions
```

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+## Relevant system information
+
+Please include:
+
+- the Rust version you're using (you can get this by running `cargo --version`)
+  - Bevy relies on the "latest stable release" of Rust
+  - nightly should generally work, but there are sometimes regressions: please let us know!
+- the operating system or browser used, including its version
+  - e.g. Windows 10, Ubuntu 18.04, iOS 14
```
Comment (mockersf) [inline .github/ISSUE_TEMPLATE/performance_regression.md:22]:
```suggestion
* the Rust version you're using (you can get this by running `cargo --version`)
  * Bevy relies on the "latest stable release" of Rust
  * nightly should generally work, but there are sometimes regressions: please let us know!
* the operating system or browser used, including its version
  * e.g. Windows 10, Ubuntu 18.04, iOS 14
```

File: .github/ISSUE_TEMPLATE/performance_regression.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+If the exact scenario is not immediately reproducible on `cargo run`, please include a set list of steps to produce the correct setup.
+
+## Before and After Traces
+
+To best help us investigate the regression, it's best to provide as much detailed profiling
+data as possible.
+
+If your app is running slowly, please show profiler traces before and after the change.
+For more information on how to get these traces, see
+https://github.com/bevyengine/bevy/blob/main/docs/profiling.md.
```
Comment (alice-i-cecile) [inline .github/ISSUE_TEMPLATE/performance_regression.md:38]:
```suggestion
<https://github.com/bevyengine/bevy/blob/main/docs/profiling.md>.
```

**Pull request merged**